oshmem/shmem/c/shmem_free.c

```diff
@@ -41,7 +42,12 @@ static inline void _shfree(void* ptr)
 {
     int rc;
 
-    RUNTIME_CHECK_INIT(); RUNTIME_CHECK_ADDR(ptr);
+    RUNTIME_CHECK_INIT();
+    if (NULL == ptr) {
+        return OSHMEM_SUCCESS;
+    }
+
+    RUNTIME_CHECK_ADDR(ptr);
 
 #if OSHMEM_SPEC_COMPAT == 1
```

Comment:
Not sure if this compatibility check is needed before returning success on shmem_free(nil). Any input on that?

Reply:
I think it's okay, but you're right we don't want null pointers going through RUNTIME_CHECK_ADDR().